pkg/kubernetes/platforminfo.go

```diff
@@ -87,3 +87,16 @@ func (platformInfo *PlatformInfo) CreatePDBObjectList() client.ObjectList {
 		return &policyv1.PodDisruptionBudgetList{}
 	}
 }
+
+func (platformInfo *PlatformInfo) GetAgentResourcesKind(withCiliumResources bool) []ObjectKind {
+	return getResourcesKind(withCiliumResources, platformInfo.supportsPSP())
+}
+
+func (platformInfo *PlatformInfo) supportsPSP() bool {
+	if platformInfo.apiOtherVersions == nil || platformInfo.apiPreferredVersions == nil {
+		return true
+	}
+	_, ok1 := platformInfo.apiOtherVersions["PodSecurityPolicy"]
+	_, ok2 := platformInfo.apiPreferredVersions["PodSecurityPolicy"]
```

Comment:
nit: can we name these `otherExists` and `preferredExists` or something clearer than ok?

Reply:
Updated.